### PR TITLE
Brigitte Abendis finished, all scarlet tech tree, tooltips, lategame quests and buildings finished

### PR DESCRIPTION
**Pull request by SebastianTanoni**

### CODE REVIEW

**Comment by YakaryBovine** [comment]
image
Can't upgrade to Keep.

**Comment by YakaryBovine** [comment]
image
Can't upgrade Keep to Castle.

**Comment by YakaryBovine** [comment]
image
Shows level 1 but there doesn't seem to be a way to get level 2.

**Comment by SebastianTanoni** [comment]
> image Shows level 1 but there doesn't seem to be a way to get level 2.

isnt it improved with the attack upgrades? i feel like thats how paladin works forever

**Comment by YakaryBovine** [comment]
> > image Shows level 1 but there doesn't seem to be a way to get level 2.
> 
> isnt it improved with the attack upgrades? i feel like thats how paladin works forever

Oh true, looks like that does work. I don't remember re-coding that in C#.

**Comment by YakaryBovine** [comment]
Ensnare says level 1 but I can't get level 2.
image

**Comment by YakaryBovine** [comment]
Can't build any buildings that require Town Hall even though I have a Town Hall.
image

**Comment by YakaryBovine** [comment]
No ships in the Shipyard.
image

**Comment by YakaryBovine** [comment]
Unholy Archon should be in the "Can learn" category. Also should be Sanctify, not Holy Fire.
image

File: mapdata/WarcraftLegacies/AbilityData/Skin/827600961.json [hunk 1]
```diff
@@ -1,50 +1,40 @@
...
-      "Value": "Objects\\Spawnmodels\\Other\\NeutralBuildingExplosion\\NeutralBuildingExplosion.mdl,war3mapImported\\HolyStrike.mdx"
+      "Value": "(Bishop)"
     },
     {
-      "Id": 2037082209,
+      "Level": 1,
+      "Id": 828536161,
       "Type": 3,
-      "Value": "W"
+      "Value": "Causes the Scarlet Bishop to transform into a Unholy Archon upon death, allowing it to continue fighting for 30 seconds."
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/827600961.json:17]:
```suggestion
      "Value": "Causes the Scarlet Bishop to transform into an Shadow Archon upon death, allowing it to continue fighting for 30 seconds. The Shadow Archon has a powerful Chaos attack and no abilities."
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/827600961.json [hunk 2]
```diff
@@ -1,50 +1,40 @@
...
     },
     {
       "Id": 1835101793,
       "Type": 3,
-      "Value": "Greater Holy Fire"
+      "Value": "Unholy Archon"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/827600961.json:22]:
```suggestion
      "Value": "Shadow Archon"
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/827600961.json [hunk 3]
```diff
@@ -1,50 +1,40 @@
...
     },
     {
       "Level": 1,
       "Id": 829453409,
       "Type": 3,
-      "Value": "Greater Holy Fire"
+      "Value": "Unholy Archon"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/827600961.json:28]:
```suggestion
      "Value": "Shadow Archon"
```

**Comment by YakaryBovine** [comment]
image
As a limited caster, this should have Resistant Skin right?

**Comment by YakaryBovine** [comment]
image
Didn't even look but I bet this building doesn't have Druid of the Scythe training.

File: mapdata/WarcraftLegacies/UnitData/Skin/1362702440.json [hunk 4]
```diff
@@ -43,12 +43,12 @@
     {
       "Id": 1651864693,
       "Type": 3,
-      "Value": "Scarlet commander.|n|cfff5962dAbilities:|R Charge, Deploy Trebuchet, United for the Cause and the Crusade|n|n|cff99b4d1Can only train 8|n|cffffcc00Attacks land units.|r"
+      "Value": "Scarlet commander.|n|cfff5962dAbilities:|R Charge, Deploy Trebuchet, United for the Cause, United for the Crusade|n|n|cff99b4d1Can only train 8|n|cffffcc00Attacks land units.|r"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/UnitData/Skin/1362702440.json:46]:
```suggestion
      "Value": "Scarlet commander.|n|cfff5962dAbilities:|R Charge, Deploy Trebuchet, United for the Cause, United for the Crusade|n|n|cff99b4d1Can only train 8.|n|cffffcc00Attacks land units.|r"
```

Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/UnitData/Skin/1362702440.json]:
```suggestion
      "Value": "Scarlet commander.|n|cfff5962dAbilities:|R Charge!, Deploy Trebuchet, United for the Cause, United for the Crusade|n|n|cff99b4d1Can only train 8.|n|cffffcc00Attacks land units.|r"
```

**Comment by YakaryBovine** [comment]
image
Not a Hippogryph Rider nor a woman! Maybe just reword this to say "this unit", and change "her" to their so it's generic.

**Comment by YakaryBovine** [comment]
image
Wrong limit, is actually 6.

**Comment by YakaryBovine** [comment]
image
Missing Critical Strike.

**Comment by YakaryBovine** [comment]
image
Righteous Fury is just Pulverize and Retribution Aura is just Thorns Aura (actually it's even named Thorns Aura. If you want to keep it like this we'll need to rewrite our guidelines.
image

**Comment by YakaryBovine** [comment]
image
This Vendor-Hall has an inventory which is pretty pointless because if you try to put things in it, they just get sold.

**Comment by YakaryBovine** [comment]
image
No point having a Control Point level requirement of 1; might as well just be an objective to capture it.
Also need to add the Crimson Cathedral part to the reward description.

**Comment by YakaryBovine** [comment]
Duration on Divine Intervention is too long, you can see the Scroll of Protection buff it gives. Probavbly just need to reduce it to 0.01: 
image

File: mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    },
+    {
+      "Id": 1952805473,
+      "Type": 3,
+      "Value": "Learn Reckless Onslaught - [|cffffcc00Level %d|r]"
+    },
+    {
+      "Id": 1953854049,
+      "Type": 3,
+      "Value": "Brigitte complete abandonnement to vengance leads her to resetting all her abilities cooldown, but reducing her life regeneration rate by 10000% for 1 seconds.|n|n|cffffcc00Level 1|r - 180 second cooldown.|n|cffffcc00Level 2|r - 140 second cooldown.|n|cffffcc00Level 3|r - 120 second cooldown.|n|cffffcc00Level 4|r - 100 second cooldown."
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json:41]:
```suggestion
      "Value": "Resets the caster's ability cooldowns, but reduces their life regeneration rate by 10000% for 1 second.|n|n|cffffcc00Level 1|r - 180 second cooldown.|n|cffffcc00Level 2|r - 140 second cooldown.|n|cffffcc00Level 3|r - 120 second cooldown.|n|cffffcc00Level 4|r - 100 second cooldown."
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+      "Level": 3,
+      "Id": 829453409,
+      "Type": 3,
+      "Value": "Reckless Onslaught - [|cffffcc00Level 3|r]"
+    },
+    {
+      "Level": 1,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Brigitte complete abandonnement to vengance leads her to resetting all her abilities cooldown, but reducing her life regeneration rate by 10000% for 1 seconds.|n|cff99b4d1180 second cooldown|r"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json:65]:
```suggestion
      "Value": "Resets the caster's ability cooldowns, but reduces their life regeneration rate by 10000% for 1 second.|n|cff99b4d1180 second cooldown|r"
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    },
+    {
+      "Level": 2,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Brigitte complete abandonnement to vengance leads her to resetting all her abilities cooldown, but reducing her life regeneration rate by 10000% for 1 seconds.|n|cff99b4d1140 second cooldown|r"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json:71]:
```suggestion
      "Value": "Resets the caster's ability cooldowns, but reduces their life regeneration rate by 10000% for 1 second.|n|cff99b4d1140 second cooldown|r"
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    },
+    {
+      "Level": 3,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Brigitte complete abandonnement to vengance leads her to resetting all her abilities cooldown, but reducing her life regeneration rate by 10000% for 1 seconds.|n|cff99b4d1120 second cooldown|r"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json:77]:
```suggestion
      "Value": "Resets the caster's ability cooldowns, but reduces their life regeneration rate by 10000% for 1 second.|n|cff99b4d1120 second cooldown|r"
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+      "Level": 4,
+      "Id": 829453409,
+      "Type": 3,
+      "Value": "Reckless Onslaught - [|cffffcc00Level 4|r]"
+    },
+    {
+      "Level": 4,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Brigitte complete abandonnement to vengance leads her to resetting all her abilities cooldown, but reducing her life regeneration rate by 10000% for 1 seconds.|n|cff99b4d1100 second cooldown|r"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json:99]:
```suggestion
      "Value": "Resets the caster's ability cooldowns, but reduces their life regeneration rate by 10000% for 1 second.|n|cff99b4d1100 second cooldown|r"
```

**Comment by YakaryBovine** [comment]
image
Requires a quest I can't complete.

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
Honestly really well done, your attention to detail with this one was really solid. Just need to test with synergy on next time.

**Comment by SebastianTanoni** [comment]
> image This Vendor-Hall has an inventory which is pretty pointless because if you try to put things in it, they just get sold.

image
It doesnt have an inventory ability

**Comment by YakaryBovine** [comment]
Wrong research requirement: 
image

**Comment by YakaryBovine** [comment]
IIRC Madsen and Techno recently did some work on making sure heroes in Altars are organized by their level in ascending order, so Sally should be placed at the end:
image

**Comment by YakaryBovine** [comment]
Same issue with the Castle upgrade: 
image

File: mapdata/WarcraftLegacies/AbilityData/Skin/1163210817.json [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+      "Level": 4,
+      "Id": 829453409,
+      "Type": 3,
+      "Value": "Rounds for Everyone - [|cffffcc00Level 4|r]"
+    },
+    {
+      "Level": 1,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Restores 99999 hit points to friendly units in a large radius."
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1163210817.json:78]:
```suggestion
      "Value": "Fully heals all friendly units in a large radius."
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1479562323.json [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    },
+    {
+      "Id": 1952805473,
+      "Type": 3,
+      "Value": "Learn Divine Strike - [|cffffcc00Level %d|r]"
+    },
+    {
+      "Id": 1953854049,
+      "Type": 3,
+      "Value": "Calls a Divine Hammer down from the sky, dealing damage and stunning enemy land units in an area.   |n|n|cffffcc00Level 1|r - \u003CST0X,DataA1\u003E damage and stunning enemy land units for \u003CST0X,Dur1\u003E seconds.|n|cffffcc00Level 2|r -  \u003CST0X,DataA2\u003E damage and stunning enemy land units for \u003CST0X,Dur2\u003E seconds  |n|cffffcc00Level 3|r -  \u003CST0X,DataA3\u003E damage and stunning enemy land units for \u003CST0X,Dur3\u003E seconds"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1479562323.json:36]:
```suggestion
      "Value": "Calls a Divine Hammer down from the sky, dealing damage and stunning enemy land units in an area.   |n|n|cffffcc00Level 1|r - \u003CST0X,DataA1\u003E damage and stunning enemy land units for \u003CST0X,Dur1\u003E seconds.|n|cffffcc00Level 2|r -  \u003CST0X,DataA2\u003E damage and stunning enemy land units for \u003CST0X,Dur2\u003E seconds. |n|cffffcc00Level 3|r -  \u003CST0X,DataA3\u003E damage and stunning enemy land units for \u003CST0X,Dur3\u003E seconds."
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+    },
+    {
+      "Id": 1952805473,
+      "Type": 3,
+      "Value": "Learn Reckless Onslaught - [|cffffcc00Level %d|r]"
+    },
+    {
+      "Id": 1953854049,
+      "Type": 3,
+      "Value": "Resets the caster\u0027s ability cooldowns, but reduces their life regeneration rate by 10000% for 1 second.|n|n|cffffcc00Level 1|r - 180 second cooldown.|n|cffffcc00Level 2|r - 140 second cooldown.|n|cffffcc00Level 3|r - 120 second cooldown.|n|cffffcc00Level 4|r - 100 second cooldown."
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json:41]:
```suggestion
      "Value": "Resets the caster\u0027s ability cooldowns, but reduces their life regeneration rate by 100 for 1 second.|n|n|cffffcc00Level 1|r - 180 second cooldown.|n|cffffcc00Level 2|r - 140 second cooldown.|n|cffffcc00Level 3|r - 120 second cooldown.|n|cffffcc00Level 4|r - 100 second cooldown."
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+      "Level": 3,
+      "Id": 829453409,
+      "Type": 3,
+      "Value": "Reckless Onslaught - [|cffffcc00Level 3|r]"
+    },
+    {
+      "Level": 1,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Resets the caster\u0027s ability cooldowns, but reduces their life regeneration rate by 10000% for 1 second.|n|cff99b4d1180 second cooldown|r"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json:65]:
```suggestion
      "Value": "Resets the caster\u0027s ability cooldowns, but reduces their life regeneration rate by 100 for 1 second.|n|cff99b4d1180 second cooldown|r"
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+    },
+    {
+      "Level": 2,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Resets the caster\u0027s ability cooldowns, but reduces their life regeneration rate by 10000% for 1 second.|n|cff99b4d1140 second cooldown|r"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json:71]:
```suggestion
      "Value": "Resets the caster\u0027s ability cooldowns, but reduces their life regeneration rate by 100 for 1 second.|n|cff99b4d1140 second cooldown|r"
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+    },
+    {
+      "Level": 3,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Resets the caster\u0027s ability cooldowns, but reduces their life regeneration rate by 10000% for 1 second.|n|cff99b4d1120 second cooldown|r"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json:77]:
```suggestion
      "Value": "Resets the caster\u0027s ability cooldowns, but reduces their life regeneration rate by 100 for 1 second.|n|cff99b4d1120 second cooldown|r"
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+      "Level": 4,
+      "Id": 829453409,
+      "Type": 3,
+      "Value": "Reckless Onslaught - [|cffffcc00Level 4|r]"
+    },
+    {
+      "Level": 4,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Resets the caster\u0027s ability cooldowns, but reduces their life regeneration rate by 10000% for 1 second.|n|cff99b4d1100 second cooldown|r"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1129590849.json:99]:
```suggestion
      "Value": "Resets the caster\u0027s ability cooldowns, but reduces their life regeneration rate by 100 for 1 second.|n|cff99b4d1100 second cooldown|r"
```

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
Just found a handful of other issues, all tiny.

**Comment by SebastianTanoni** [comment]
> Just found a handful of other issues, all tiny.

Just fixed them and pushed!

**Review by YakaryBovine — APPROVED** [review]
Looks great!
